### ISSUE TITLE
Validation of similar paths produce error on required parameters

### ISSUE THREAD

**Issue by andrejflorjancic** [kogosoftwarellc/open-api#663]
Framework: express
openapi: 6.0.0
swagger: 2.0

When using paths validation has problems on similar paths.

Example:
```
<project>
        paths/
               users/
                     {id}.js
               articles/
                     {id}.js
```
**users/{id}.js:**
```
module.exports.get.apiDoc = {
  description: 'A description for retrieving a user.',
  tags: ['users'],
  operationId: 'getUser',
  // parameters for this operation
  parameters: [
    {
      in: 'path',
      name: 'id',
      type: 'string'
    }, {
      in: 'query',
      name: 'country',
      required: true,
      type: 'string'
    }
  ],
  responses: {
    default: {
      $ref: '#/definitions/Error'
    }
  }
};
```
**articles/{id}.js:**
```
module.exports.get.apiDoc = {
  description: 'A description for retrieving a article.',
  tags: ['articles'],
  operationId: 'getArticle',
  // parameters for this operation
  parameters: [
    {
      in: 'path',
      name: 'id',
      type: 'string'
    }, {
      in: 'query',
      name: 'store',
      required: true,
      type: 'string'
    }
  ],
  responses: {
    default: {
      $ref: '#/definitions/Error'
    }
  }
};
```
Calling ../users/{id} will produce error **should have required property 'store'**.

Renaming path **articles** to **xarticles** and than calling ../xarticles/{id} will produce error **should have required property 'country'**.

**Comment by malnor** [comment]
Been sitting with this for a day now. Have a very similar problem /clients GET getting mixed up with /client/{client_id} GET.

**Comment by malnor** [comment]
@andrejflorjancic after debugging the code I found that the problem was actually my code and how I attached the apiDoc to the object that caused problems. module.exports.get.apiDoc = xxx; isn't the best way of doing this.

So this can be ignored. Even though this is actually more of a javascript problem, the documentation could be more clear on having the same method multiple times requires a different pattern. It may be possible to improve the code to avoid this also, but I'm not sure on that.
